### PR TITLE
Attach sources when deploying or preparing a release

### PR DESCRIPTION
**Pull request by lukaszlenart**
**NOTE**: this fix didn't help (at least when testing locally)

Each time I'm trying to prepare a new release I must manually create and deploy source by checking out a tag and then running `mvn source:jar deploy -Prelease` again.

Even with this fix, running `mvn deploy` doesn't produce a jar file with source code locally.

Any help? cc: @sepe81 @sdutry

### CODE REVIEW

**Comment by sepe81** [comment]
@lukaszlenart Do you use the maven-release-plugin via [mvn release:perform](https://maven.apache.org/maven-release/maven-release-plugin/perform-mojo.html#release-perform) for releases? If so, could you share the exact command you use?

Instead of using `<arguments>-Prelease ${arguments}</arguments>`, you might consider `<releaseProfiles>release</releaseProfiles>`, which [appears to be more explicit](https://maven.apache.org/maven-release/maven-release-plugin/perform-mojo.html#releaseprofiles).

Perhaps using `<dryRun>true</dryRun>` could also be helpful for debugging?

File: pom.xml [hunk 1]
```diff
@@ -189,6 +189,15 @@
 					<!-- Attach source jar to builds -->
 					<artifactId>maven-source-plugin</artifactId>
 					<version>${maven-source-plugin.version}</version>
+					<executions>
+						<execution>
+							<id>attach-sources</id>
+							<phase>verify</phase>
```
Comment (sepe81) [inline pom.xml]:
```suggestion
```

The default is `package`, which should be sufficient.

see https://maven.apache.org/plugins/maven-source-plugin/jar-no-fork-mojo.html

**Review by sepe81 — CHANGES_REQUESTED** [review]

**Comment by sepe81** [comment]
I adjusted the plugin order locally and observed that source:3.3.1:jar-no-fork now runs before deploy. Hopefully, this resolves the issue. The plugin execution order in Maven can be tricky at times.

[INFO] --- javadoc:3.11.1:jar (attach-source) @ struts2-jquery ---
[INFO] Not executing Javadoc as the project is not a Java classpath-capable package
[INFO] 
[INFO] --- source:3.3.1:jar-no-fork (attach-sources) @ struts2-jquery ---
[INFO] 
[INFO] --- install:3.1.2:install (default-install) @ struts2-jquery ---
[INFO] Installing /Users/speters/workspace/struts2-jquery/pom.xml to /Users/speters/.m2/repository/com/jgeppert/struts2/jquery/struts2-jquery/5.1.0-SNAPSHOT/struts2-jquery-5.1.0-SNAPSHOT.pom
[INFO] 
[INFO] --- deploy:3.1.3:deploy (default-deploy) @ struts2-jquery ---


image

**Comment by lukaszlenart** [comment]
> `<releaseProfiles>release</releaseProfiles>`

Looks like that was it, thanks!

**Review by sepe81 — APPROVED** [review]

**Pull request merged**